app/views/edit/build-config.html

```diff
@@ -141,254 +165,278 @@
                               <div>
                                 <input class="form-control"
                                   type="text"
-                                  ng-model="imageOptions.from.dockerImage"
+                                  ng-model="imageOptions.fromSource.imageStreamImage"
+                                  placeholder="example: openshift/ruby-20-centos7@603bfa418"
                                   autocorrect="off"
                                   autocapitalize="off"
-                                  placeholder="example: centos/ruby-20-centos7:latest"
                                   spellcheck="false"
                                   required>
                               </div>
                             </div>
 
-                            <div ng-if="imageOptions.from.type==='ImageStreamImage'" class="form-group">
-                              <label for="FromTypeImage">Image Stream Image</label>
+
+                            <div ng-if="imageOptions.fromSource.type==='DockerImage'" class="form-group">
+                              <label for="imageSourceLink">Docker Image Repository</label>
                               <div>
                                 <input class="form-control"
+                                  id="imageSourceLink"
+                                  name="imageSourceLink"
                                   type="text"
-                                  ng-model="imageOptions.from.imageStreamImage"
-                                  placeholder="example: openshift/ruby-20-centos7@603bfa418"
+                                  ng-model="imageOptions.fromSource.dockerImage"
+                                  placeholder="example: centos/ruby-20-centos7:latest"
                                   autocorrect="off"
                                   autocapitalize="off"
                                   spellcheck="false"
                                   required>
                               </div>
                             </div>
-                          </div>
 
-                          <div class="form-group">
-                            <div class="checkbox">
-                              <label>
-                                <input type="checkbox" ng-model="options.forcePull"/>
-                                Always pull the builder image from the docker registry, even if it is present locally
-                              </label>
+                            <div class="form-group">
+                              <label for="buildFrom">Source and Destination Paths<span class="help action-inline">
+                                <a href="">
+                                <i class="pficon pficon-help" data-toggle="tooltip" aria-hidden="true" data-original-title="Paths is a list of source and destination paths to copy from the image. At least one pair has to be specified."></i>
```

Comment:
"List of source and destination paths to copy from the image. At least one pair has to be specified."